cupy/random/_generator_api.pyx

```diff
@@ -262,6 +265,55 @@ class Generator:
         """
         return self.standard_exponential(size) * scale
 
+    def geometric(self, p, size=None):
+        """Geometric distribution.
+
+        Returns an array of samples drawn from the geometric distribution. Its
+        probability mass function is defined as
+
+        .. math::
+            f(x) = p(1-p)^{k-1}.
+
+        Args:
+            p (float or array_like of floats): Success probability of
+                the geometric distribution.
+            size (int or tuple of ints): The shape of the array. If ``None``, a
+                zero-dimensional array is generated.
+
+        Returns:
+            cupy.ndarray: Samples drawn from the geometric distribution.
+
+        .. seealso::
+            :func:`numpy.random.Generator.geometric`
+        """
+        cdef ndarray y
+        cdef ndarray p_arr
+
+        if not isinstance(p, ndarray):
+            if type(p) in (float, int):
+                p_a = ndarray((), numpy.float64)
+                p_a.fill(p)
+                p = p_a
+            else:
+                raise TypeError('p is required to be a cupy.ndarray'
+                                ' or a scalar')
+        else:
+            # Check if size is broadcastable to shape
+            # but size determines the output
+            p = p.astype('d', copy=False)
```

Comment:
Can we remove this test, as due to this test input array is required to be cupy.ndarray. So something like `rng.geometric([0.2, 0.3, 0.4])` will not work, but was supported by numpy.

Reply:
Making p take array-like other than cupy.ndarray is okay.

Reply:
we only allow cupy arrays or scalars in order to avoid implicit synchronization.